### DISCUSSION TITLE
Bind command to property's method?

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#241] [Q&A]
Is it possible to bind command to viewmodel's **property's** method, instead of viewmodel's method?  For example:

```
<Button HorizontalAlignment="Right" Content="Add Port" Command="{s:Action Ports.AddPort}"/>
```

where `Ports` is a viewmodel property who has its own method `AddPort`.  

I've tried and it failed, but I'd like to confirm it won't work before I go ahead and create every wrapper method in viewmodel.  I surely hope it will work as creating dozens of wrapper methods isn't that fun!  Thanks in advance!

**Comment by canton7** [comment]
You can do it by changing the element's ActionTarget:

```xaml
<Button Content="Add Port" s:View.ActionTarget="{Binding Ports}" Command="{s:Action AddPort}"/>
```

You can set the ActionTarget on a parent element and it will be inherited by all child elements, if that helps

**Reply by mwdpb** [reply]
That works like a charm!  Thank you very much for the help!
